src/topp/OpenSwathAssayGenerator.cpp

```diff
@@ -217,14 +284,33 @@ class TOPPOpenSwathAssayGenerator :
       }
       else {uis_swathes = swathes;}
       
-      std::cout << "Generating identifying (UIS) transitions" << std::endl;
+      LOG_INFO << "Generating identifying transitions for IPF" << std::endl;
       assays.uisTransitions(targeted_exp, allowed_fragment_types, allowed_fragment_charges, enable_identification_specific_losses, enable_identification_unspecific_losses, enable_identification_ms2_precursors, product_mz_threshold, uis_swathes, -4, max_num_alternative_localizations, -1);
       std::vector<std::pair<double, double> > empty_swathes;
       assays.restrictTransitions(targeted_exp, product_lower_mz_limit, product_upper_mz_limit, empty_swathes);
     }
 
-    std::cout << "Writing assays " << out << std::endl;
-    traml.store(out, targeted_exp);
+    LOG_INFO << "Writing assays " << out << std::endl;
+    if (out_type == FileTypes::TSV)
+    {
+      const char* tr_file = out.c_str();
+      TransitionTSVReader tsv_reader = TransitionTSVReader();
+      tsv_reader.setLogType(log_type_);
+      tsv_reader.convertTargetedExperimentToTSV(tr_file, targeted_exp);
+    }
+    if (out_type == FileTypes::PQP)
+    {
+      const char * tr_file = out.c_str();
+      TransitionPQPReader pqp_reader = TransitionPQPReader();
+      pqp_reader.setLogType(log_type_);
+      pqp_reader.convertTargetedExperimentToPQP(tr_file, targeted_exp);
```

Comment:
this reads a bit strange. we have a reader that is also a PQP writer

Reply:
Good point. Originally, we provided read-only support via TransitionTSVReader. We could potentially promote both TransitionTSVReader and TransitionPQPReader to a real format in OpenMS. Since other open PR would be affected as well, I suggest we do this in a separate PR.

Reply:
fine with me